core/changedetector/proto_change_detector_service.go

```diff
@@ -0,0 +1,215 @@
+		return nil, errors.New("unable getSchemaChangeEvent get file descriptor set from previous schema data")
+	}
+	setDirectlyImpactedSchemasAndFields(currentFds, prevFds, sce)
+	reverseDependencies := make(map[string][]string)
+	populateReverseDependencies(currentFds, reverseDependencies)
+	for _, schema := range sce.UpdatedSchemas {
+		appendImpactedDependents(sce, schema, findDependentImpactedSchemas(reverseDependencies, schema))
+	}
+	return sce, nil
+}
+
+func appendImpactedDependents(sce *stencilv1beta2.SchemaChangedEvent, key string, impactedDependents []string) {
+	if sce.ImpactedSchemas == nil {
+		sce.ImpactedSchemas = make(map[string]*stencilv1beta2.ImpactedSchemas)
+	}
+	sce.ImpactedSchemas[key] = &stencilv1beta2.ImpactedSchemas{
+		SchemaNames: impactedDependents,
+	}
+}
+
+func setDirectlyImpactedSchemasAndFields(currentFds, prevFds *descriptor.FileDescriptorSet, sce *stencilv1beta2.SchemaChangedEvent) {
+	packageMessageMap := getPackageMessageMap(prevFds)
+	packageEnumMap := getPackageEnumMap(prevFds)
+	for _, fds := range currentFds.GetFile() {
+		for _, newMessageDesc := range fds.GetMessageType() {
+			messageName := fds.GetPackage() + "." + newMessageDesc.GetName()
+			if oldMessageDesc := getMessageDescriptor(packageMessageMap, fds.GetPackage(), newMessageDesc.GetName()); oldMessageDesc != nil {
+				if !proto.Equal(newMessageDesc, oldMessageDesc) {
+					sce.UpdatedSchemas = append(sce.UpdatedSchemas, messageName)
+					appendImpactedFields(sce, messageName, getImpactedMessageFields(oldMessageDesc, newMessageDesc))
+				}
+				for _, newEnumDesc := range newMessageDesc.GetEnumType() {
+					enumName := messageName + "." + newEnumDesc.GetName()
+					if oldEnumDesc := findEnumDescriptorFromMessageDescriptor(oldMessageDesc, newEnumDesc.GetName()); oldEnumDesc != nil {
+						if !proto.Equal(newEnumDesc, oldEnumDesc) {
+							sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
+							appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
+						}
+					} else {
+						sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
+						appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
+					}
+				}
+			} else {
+				sce.UpdatedSchemas = append(sce.UpdatedSchemas, messageName)
+				appendImpactedFields(sce, messageName, getImpactedMessageFields(oldMessageDesc, newMessageDesc))
+			}
+		}
+		compareEnumDescriptors(fds, packageEnumMap, sce)
+	}
+}
+func appendImpactedFields(sce *stencilv1beta2.SchemaChangedEvent, key string, impactedFields []string) {
+	if sce.UpdatedFields == nil {
+		sce.UpdatedFields = make(map[string]*stencilv1beta2.ImpactedFields)
+	}
+	if val, ok := sce.UpdatedFields[key]; ok {
+		val.FieldNames = append(val.FieldNames, impactedFields...)
+	} else {
+		sce.UpdatedFields[key] = &stencilv1beta2.ImpactedFields{
+			FieldNames: impactedFields,
+		}
+	}
```

Comment:
In Go, we mostly try to avoid using `else` as much as possible, to avoid one more cyclomatic complexity.

this can be easily done by adding simple `continue` or `return` in all the cases. 
```suggestion
	if val, ok := sce.UpdatedFields[key]; ok {
		val.FieldNames = append(val.FieldNames, impactedFields...)
		return
	}
       sce.UpdatedFields[key] = &stencilv1beta2.ImpactedFields{
	        FieldNames: impactedFields,
       }
	
```

Reply:
It makes sense but it depends on the requirement. Here adding else will add more readablity than return.
If we have case like this then return makes more sense.
func(isAdmin bool){
if(isAdmin){
return;
}
}

Reply:
updated

Reply:
thanks, resolving it